kt/kt.go

```diff
@@ -206,7 +208,56 @@ func NewConnTLS(host string, port int, poolsize int, timeout time.Duration, cred
 
 // NewConn creates a connection to an Kyoto Tycoon endpoint.
 func NewConn(host string, port int, poolsize int, timeout time.Duration) (*Conn, error) {
-	return newConn(host, port, poolsize, timeout, "")
+	parts := strings.Split(host, "://")
+
+	if len(parts) == 1 {
+		return newConn(host, port, poolsize, timeout, "")
+	} else if len(parts) == 2 {
+		if parts[0] != "unix" {
+			return nil, errors.New("NewConn: Unexpected network")
+		}
+		return newUnixConn(parts[1], poolsize, timeout)
+	} else {
+		return nil, errors.New("NewConn: Wrong parameters")
+	}
+}
+
+type unixDialer struct {
+	net.Dialer
+}
+
+func (d *unixDialer) Dial(network, address string) (net.Conn, error) {
+	// Hackery to remove port 80
+	parts := strings.Split(address, ":")
+	return d.Dialer.Dial("unix", parts[0])
+}
+
+func newUnixConn(socket string, poolsize int, timeout time.Duration) (*Conn, error) {
+	c := Conn{
+		scheme:  "http",
+		timeout: timeout,
+		host:    socket,
+		transport: &http.Transport{
+			ResponseHeaderTimeout: timeout,
+			IdleConnTimeout:       30 * time.Second,
+
+			Dial: (&unixDialer{net.Dialer{
+				Timeout:   30 * time.Second,
```

Comment:
I think it will be good option to use timeout argument instead of hardcoding 30 seconds.